src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/Couchbase3/Instrumentation.xml

```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+Copyright 2020 New Relic Corporation. All rights reserved.
+SPDX-License-Identifier: Apache-2.0
+-->
+<extension xmlns="urn:newrelic-extension">
+  <instrumentation>
+
+    <tracerFactory name="Couchbase3CollectionWrapper">
```

Comment:
Instrumenting each method is probably an ok short term approach. However we may want to consider leveraging the functionality called out in https://docs.couchbase.com/dotnet-sdk/current/howtos/observability-tracing.html to make things easier going forward. That can be done in a separate follow up ticket.

Reply:
Feel free to create that ticket. I looked at that functionality briefly and couldn't see an obvious way we could instrument it.

Reply:
I don't think that we'd instrument it. I think we'd need to do something similar to what we do for newer Redis libraries where we integrate with the profiling API early on, and that profiling API integration would be responsible for starting and ending the segments.